PURL-TYPES.rst

```diff
@@ -255,6 +255,37 @@ rpm
       pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
       pkg:rpm/centerim@4.22.10-1.el6?arch=i686&epoch=1&distro=fedora-25
 
+swift
+----
+``swift`` for Swift packages:
+
+- There is currently no default repository for Swift packages.
+- The ``namespace`` is used for the scope of a Swift package.
+  It is not case sensitive and must be lowercased.
```

Comment:
Is namespace really not case sensitive? Take https://swiftpackageregistry.com/SDWebImage/SDWebImage as an example. To my knowledge SDWebImage is both the namespace and name and both have upper and lower case letters. Also to my knowledge, most Swift packages use GitHub which is not case sensitive. But Swift is decentralized and could leverage alternative services which could introduce case sensitivity, correct? Or is my understanding wrong?
 
 Also a question... GitHub uses organizations (or users) and these essentially become the namespace. However, given that Swift packages could be hosted elsewhere, is namespace a requirement? I haven't seen any indication that namespace is required. If this is true, then the spec should state that namespace is optional.

Reply:
A package's namespace is case-insensitive and case-preserving. In the case of your example, the canonical namespace is "SDWebImage", and that's what would be returned in JSON responses from the server. However, the namespace is case-insensitive, meaning that the server doesn't distinguish among "sdwebimage", "SDWebImage", or any other strings that differ only in case. This is similar to how GitHub handles URLs to repositories. github.com/sdwebimage/sdwebimage resolves the same as github.com/SDWebImage/SDWebImage